lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -1338,6 +1369,47 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
     }
   }
 
+  void _handleSubaddressingAction(MailboxId mailboxId, Map<String, List<String>?>? currentRights, MailboxActions subaddressingAction) {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final session = mailboxDashBoardController.sessionCurrent;
+
+    if (session != null && accountId != null) {
+      final allowSubaddressingRequest = SubaddressingRequest(
+          mailboxId,
+          currentRights,
+          subaddressingAction == MailboxActions.allowSubaddressing ? MailboxSubaddressingAction.allow : MailboxSubaddressingAction.disallow
+      );
+
+      consumeState(_subaddressingInteractor.execute(session, accountId, allowSubaddressingRequest));
+    } else {
+      _handleSubaddressingFailure(SubaddressingFailure.withException(
+          SubaddressingException("Session and accountId should not be null")));
```

Comment:
Should create specific class for this exception

Reply:
done !